drivers/gpio/gpio_imx.c

```diff
@@ -82,6 +82,9 @@ static int imx_gpio_configure(const struct device *port, gpio_pin_t pin,
 
 	/* Init pin configuration struct, and use pinctrl api to apply settings */
 	__ASSERT_NO_MSG(pin < config->mux_count);
+
+	unsigned int key = irq_lock();
```

Comment:
Thanks for your review, that's true, the memcpy occurs on a local stack allocated struct, so it is indeed safe. I'll move the irq_lock just before the pin_cfg.pin_ctrl_flags since everything needs to be coherent when we setting up pinmux/ctrl.

Reply:
Fixed in latest submit :)